sci-mathematics/cgal/cgal-5.2-r1.ebuild

```diff
@@ -57,7 +55,7 @@ src_configure() {
 	local mycmakeargs=(
 		-DCGAL_INSTALL_LIB_DIR="$(get_libdir)"
 		-DCGAL_INSTALL_CMAKE_DIR="$(get_libdir)/cmake/CGAL"
-		-DCGAL_HEADER_ONLY=OFF
+		-DCGAL_HEADER_ONLY=ON
```

Comment:
why? and why are you doing this in one commit?

Reply:
It's the default setting, we have no specific need for making it into a library. From https://doc.cgal.org/latest/Manual/installation.html
> Since CGAL version 5.0, CGAL is header-only be default, which means that there is no need to compile CGAL or its libraries before it can be used.
>
> This page is for advanced users that have a good reason to still use the old way. If this is not your case, head over back to the page Getting Started with CGAL. 

I'll split it into bump and header only changes.

Reply:
Thanks for clearing that up. Here's the plan: Change it for 5.2.1 only, and in due course the old ebuilds will be washed out of the repo. I don't want to change stable ebuilds like this.

Reply:
done.

Reply:
This change breaks other packages in the tree which need this library:
https://bugs.gentoo.org/791811

Reply:
These packages should be patched to support the new CGAL.
Or we can restrict the CGAL to older version.
Will do this soon.

Reply:
> These packages should be patched to support the new CGAL.
> Or we can restrict the CGAL to older version.
> Will do this soon.

You should add a USE flag to control whether library is needed or not and let users to decide. This will also allow to keep current packages working without patching other than dependency update.

Reply:
@epsilon-0 thanks for looking into it. Please either ping me in the PR with fixes to related packages, or and preferably, link the commits to this bug:
https://bugs.gentoo.org/791811

Reply:
Indeed, it seems like the simplest solution is to allow a `shared` USE flag. I've made a PR to do a version bump and add the USE flag - https://github.com/gentoo/gentoo/pull/21275
I am unsure what are all the packages that need the library, so I haven't touched them.

Overall, the better strategy would be to tell upstreams of these packages to work with the newer CGAL.